app/code/Magento/InventoryCatalog/etc/config.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Store:etc/config.xsd">
```

Comment:
Sorry, @sinisa86  that's probably my fault, that I did not make all the investigations regarding this task in advance, but looks like we already have Stock configuration option we could just re-use.

`Decrease Stock When Order is Placed` - `cataloginventory/options/can_subtract`
http://devdocs.magento.com/guides/v2.2/config-guide/prod/config-reference-most.html
These configuration values are availabe in the `Magento Admin in Stores > Settings > Configuration > Catalog > Inventory`

This config option already used in `RegisterProductSales` function, invoked at checkout time

```
    public function registerProductsSale($items, $websiteId = null)
    {
       // ...
            $canSubtractQty = $stockItem->getItemId() && $this->canSubtractQty($stockItem);
            if (!$canSubtractQty || !$this->stockConfiguration->isQty($lockedItemRecord['type_id'])) {
                continue;
            }
            if (!$stockItem->hasAdminArea()
                && !$this->stockState->checkQty($productId, $orderedQty, $stockItem->getWebsiteId())
            ) {
                $this->getResource()->rollBack();
                throw new \Magento\Framework\Exception\LocalizedException(
                    __('Not all of your products are available in the requested quantity.')
                );
            }
            if ($this->canSubtractQty($stockItem)) {
                $stockItem->setQty($stockItem->getQty() - $orderedQty);
            }
```
Thus, no need to introduce new configuration